---
layout: subsite-galaxy
---


<br/>
<img src="/assets/media/nanogalaxy_logo.png" height="100px" alt="NanoGalaxy logo"/>

Welcome to **NanoGalaxy**  -- a webserver to process, analyse and visualize Oxford Nanopore Technologies (ONT) data and similar long-reads technologies.


1. TOC
{:toc}


# Get started

Are you new to Galaxy, or returning after a long time, and looking for help to get started? Take [a guided tour](https://nanopore.usegalaxy.eu/tours/core.galaxy_ui){:target="_blank"} through Galaxy's user interface.

Check also the standard but customizable [workflows](#workflows) available there.

# Tools

A collection of best practice and popular tools are integrated (and are expanding) in this custom Galaxy instance. The ONT-oriented and -specific subset includes:

- **Polishing, QC and preprocessing**
    - Nanopolish, Porechop, Filtlong, Poretools
- **Genome assembly** 
    - Flye, Unicycler, Wtdbg2, Miniasm, Racon, Canu
- **Mapping**
    - Minimap2, GraphMap
- **Visualisation**
    - Nanoplot, Bandage
- **Taxonomy and metagenomics**
    - Kraken2, PlasFlow, Staramr

# Tutorials

We are passionate about training. So we are working in close collaboration with the [Galaxy Training Network (GTN)](https://galaxyproject.org/teach/gtn/){:target="_blank"} to develop training materials of data analyses based on Galaxy {% cite batut2017community %}. The NanoGalaxy training materials are under development and will be soon hosted on the GTN GitHub repository are available online at [https://training.galaxyproject.org](https://training.galaxyproject.org){:target="_blank"}.


# Workflows

To orchestrate tools and help users with their analyses, some best practice workflows are prepared and become available. The workflows can be extended using similar and alternative combinations using the workflow editor.

The workflows are available in the [Shared Workflows](https://nanopore.usegalaxy.eu/workflows/list_published), with the label "***ONT***".

