---
tags: [tools]
title: Minimap2 mapper for noisy long read sequences (e.g. Nanopore, PacBio)
---

Do you have Nonopaore or PacBio reads? Try minimpa2 as a new aligner!
From minimaps [homepage](https://github.com/lh3/minimap2):

Minimap2 is a versatile sequence alignment program that aligns DNA or mRNA sequences against
a large reference database. Typical use cases include: (1) mapping PacBio or Oxford Nanopore
genomic reads to the human genome; (2) finding overlaps between long reads with error rate up
to ~15%; (3) splice-aware alignment of PacBio Iso-Seq or Nanopore cDNA or Direct RNA
reads against a reference genome; (4) aligning Illumina single- or paired-end reads;
(5) assembly-to-assembly alignment; (6) full-genome alignment between two closely
related species with divergence below ~15%.

For ~10kb noisy reads sequences, minimap2 is tens of times faster than
mainstream long-read mappers such as BLASR, BWA-MEM, NGMLR and GMAP.
It is more accurate on simulated long reads and produces biologically meaningful
alignment ready for downstream analyses. For >100bp Illumina short reads, minimap2 is three
times as fast as BWA-MEM and Bowtie2, and as accurate on simulated data.
Detailed evaluations are available from the minimap2 preprint.

Read more in the minimap2 preprint from Heng Li - [Minimap2: fast pairwise alignment for long nucleotide sequences.](https://arxiv.org/abs/1708.01492)

## Credits

Marius van den Beek for developing the Galaxy integration!
